datafusion/spark/src/function/math/ceil.rs

```diff
@@ -0,0 +1,461 @@
+            UInt16 => scale_array
+                .as_primitive::<arrow::datatypes::UInt16Type>()
+                .value(0) as i32,
+            UInt32 => scale_array
+                .as_primitive::<arrow::datatypes::UInt32Type>()
+                .value(0) as i32,
+            UInt64 => scale_array
+                .as_primitive::<arrow::datatypes::UInt64Type>()
+                .value(0) as i32,
+            other => {
+                return exec_err!("scale parameter must be an integer, got {:?}", other)
+            }
+        };
+        Some(s)
+    } else {
+        None // No scale provided
+    };
+
+    // Perform ceiling operation based on data type and scale parameter
+    match (args[0].data_type(), scale) {
+        // Float32 without scale: ceil to nearest integer, return as Int64
+        (Float32, None) => {
+            let array = value_array
+                .as_primitive::<arrow::datatypes::Float32Type>()
+                .unary::<_, arrow::datatypes::Int64Type>(|value: f32| {
+                value.ceil() as i64
+            });
+            Ok(Arc::new(array))
+        }
+        // Float32 with scale: ceil to specified decimal places, return as Float32
+        (Float32, Some(s)) => {
+            let scale_factor = 10_f32.powi(s); // 10^scale for decimal place adjustment
+            let array = value_array
+                .as_primitive::<arrow::datatypes::Float32Type>()
+                .unary::<_, arrow::datatypes::Float32Type>(|value: f32| {
+                (value * scale_factor).ceil() / scale_factor // Scale, ceil, then unscale
+            });
+            Ok(Arc::new(array))
+        }
+        // Float64 without scale: ceil to nearest integer, return as Int64
+        (Float64, None) => {
+            let array = value_array
+                .as_primitive::<arrow::datatypes::Float64Type>()
+                .unary::<_, arrow::datatypes::Int64Type>(|value: f64| {
+                value.ceil() as i64
+            });
+            Ok(Arc::new(array))
+        }
+        // Float64 with scale: ceil to specified decimal places, return as Float64
+        (Float64, Some(s)) => {
+            let scale_factor = 10_f64.powi(s); // 10^scale for decimal place adjustment
+            let array = value_array
+                .as_primitive::<arrow::datatypes::Float64Type>()
+                .unary::<_, arrow::datatypes::Float64Type>(|value: f64| {
+                (value * scale_factor).ceil() / scale_factor // Scale, ceil, then unscale
+            });
+            Ok(Arc::new(array))
+        }
+        // Int64: integers are already "ceiled", return unchanged regardless of scale
+        (Int64, None) => Ok(Arc::clone(&args[0])),
+        (Int64, Some(_)) => Ok(Arc::clone(&args[0])),
+        // Decimal128: handle decimals with scale > 0 (fractional part exists)
+        (Decimal128(precision, value_scale), scale_param) => {
+            if *value_scale > 0 {
+                match scale_param {
```

Comment:
Either this needs to be extracted into a separate function, or the checks should be pushed into the match arm, e.g.

```rust
(Decimal128(precision, value_scale), None) if *value_scale > 0 => {
```

As it currently is, it is far too indented